k8s/src/main/scala/io/buoyant/k8s/EventLogging.scala

```diff
@@ -7,7 +7,7 @@ private[k8s] trait EventLogging {
   @inline private[this] def logAction[A](verb: String, descriptor: => String, format: A => String)(value: A): Unit =
     log.ifTrace(s"k8s ns $ns service $srv $verb $descriptor ${format(value)}")
 
-  protected def logActions[A](verb: String, descriptor: => String, format: A => String)(values: Iterable[A]): Unit =
+  protected def logActions[A](verb: String, descriptor: => String, format: A => String)(values: => Iterable[A]): Unit =
     if (values.nonEmpty) {
```

Comment:
values is immediately evaluated here regardless of log level so I don't think we benefit at all from passing by name.  Ideally we would avoid evaluating unless the log level was high enough.

Reply:
To make matters worse, we actually evaluate `values` twice, incurring the cost of the set operations twice.

Reply:
Ah, you're right – I think I'll have to rewrite the whole event logger. Which is fine, this implementation is kind of ossified.

Reply:
should we close this PR then?

Reply:
I'm working on _actually_ fixing this, but would be fine with closing this PR & opening a new one?